Scripts/Singletons/PuzzleElementPlacement.gd

```diff
@@ -337,6 +342,17 @@ func call_loop_reconnections(type, child):
 			codeBlock.connect_to_LoopBlock()
 	
 
+#Allows progress checks to work after Restart, progress_check_FBA elements need to point to pre-Restart FBAs
+func init_progress_check_FBA():
+	for i in range(level.progress_check_FBA.size()):
+		if level.progress_check_FBA[i].get_parent().name == "Main":
+			level.progress_check_FBA[i] = level.get_node("IDE/Main/FunctionBlockArea")
+		elif level.progress_check_FBA[i].get_parent().name == "F1":
+			level.progress_check_FBA[i] = level.get_node("IDE/F1/FunctionBlockArea")
+		elif level.progress_check_FBA[i].get_parent().name == "F2":
+			level.progress_check_FBA[i] = level.get_node("IDE/F2/FunctionBlockArea")
+	
```

Comment:
I'm crying you really said let me hard code this real quick. This will also need to be updated with the loop and if blocks later on

Reply:
We should allow up to 3 func blocks, 3 if-else blocks, and 3 loop blocks. Also is there a better way to do this or nah

Reply:
mi hoy minoy